### PR TITLE
Add duplicate handling of tag names and further error handling

### PR DESCRIPTION
**Pull request by ywllowsensor**
Added duplicate handling for the scenarios below.
Note: All tags are case sensitive.
- When all tag(s) as inputted by the user exactly matches tags in the existing contact
- When some tags inputted by the user matches some tags in the existing contact

Added further error handling for the scenarios below.
- When the name as inputted by the user does not match any names in the address book

### CODE REVIEW

**Comment by ywllowsensor** [comment]
Open to suggestions on writing unit tests for the handleDuplicateTags method

**Comment by ywllowsensor** [comment]
Fixes #68

File: src/main/java/seedu/address/logic/commands/TagAddCommand.java [hunk 1]
```diff
@@ -77,16 +87,66 @@ public boolean equals(Object other) {
...
+                Messages.tagSetToString(tagsToAdd));
+    }
+
+    /**
+     * Handles duplicate tags as inputted by the user.
+     * Checks if any of the tags from user input matches existing tags in the
+     * original Person.
+     * Matching tags are added to a separate set that keeps track of duplicate tags.
+     * Unique tags are added to a separate list of tags.
+     * @param person EditedPerson from the execute method above.
```
Comment (siriousguy) [inline src/main/java/seedu/address/logic/commands/TagAddCommand.java:130]: Leave a space between descriptor and parameter section (as per Java convention)

**Comment by siriousguy** [comment]
LGTM! Just have one comment for a minor style fix, otherwise alls good!

File: src/main/java/seedu/address/logic/commands/TagAddCommand.java (hunk 1 above)
Comment (ywllowsensor) [reply src/main/java/seedu/address/logic/commands/TagAddCommand.java:130]: Noted RSM. It wont happen again.

**Pull request merged**